Node/core/src/bots/RestfulConnector.ts

```diff
@@ -0,0 +1,212 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+//
+// Microsoft Bot Framework: http://botframework.com
+//
+// Bot Builder SDK Github:
+// https://github.com/Microsoft/BotBuilder
+//
+// Copyright (c) Microsoft Corporation
+// All rights reserved.
+//
+// MIT License:
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED ""AS IS"", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+import readline = require('readline');
+import ub = require('./UniversalBot');
+import mb = require('../Message');
+import utils = require('../utils');
+let restify = require('restify');    //let instead of import because there's no typings for restify
+
+interface IOutMessageDictionary {
+    [key: string]: Array<IMessage>;
+};
+
+export class RestfulConnector implements ub.IConnector {
+    private handler: (events: IEvent[], cb?: (err: Error) => void) => void;
+    private rl: readline.ReadLine;
+    private replyCnt = 0;
+
+    private server = restify.createServer();
```

Comment:
Making server public will allow for bot file to add any other components required.